java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/api/ExplorerManagerFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.explorer.api;
+
+import java.util.concurrent.CompletionStage;
+import org.openide.explorer.ExplorerManager;
+import org.openide.util.Lookup;
+
+/**
+ * Factory that produces an ExplorerManager that serves node in a specified tree structure. The
+ * Factory has to be registered as a named service on path {@code Explorers/&lt;id>}; first registration
+ * wins.
+ * <p>
+ * Note: this interface will become an API independent of LSP.
```

Comment:
Great. Include steps & references to what needs to be done, please.

Reply:
Steps described in javadoc.